doc/semantics.rst

```diff
@@ -46,16 +46,42 @@ Opposed to that, components maintained by a third party are declared through the
 Identities and Accessors
 ------------------------
 
-Sources and Resources are unambiguously identified by two-tuples of name and version, in the
-context of their declaring component version.
+Component versions are unambiguously identified by two-tuples of `name` and `version` in the
+context of a `Component Descriptor Context Repository`. Their identity never changes across
+any replication between context repositories.
+
+Components declare dependencies towards `Sources`, `Resources` and other `Components`.
+
+Any of those dependency declarations have an umambiguous identity in the context of their
+declaring component version. Their identities always consist of their formal type
+(`Source`, `Resource`, `Component Reference`) and a `name`.
+
+Depending on the formal type, different restrictions for valid names are defined.
+
+Dependency declarations *MAY* declare additional identity attributes.
+
+Considering the tuple of `formal type`, `name`, and the optional additional identity attributes,
+all dependency declarations *MUST* be unique for any given component version starting with the next
+component descriptor version to be released after `v2`.
```

Comment:
should we already name the next version as `v2.1`?

Reply:
we _might_. however, as we did not yet decide on the name, I took the cheap route, and referred to it in an abstract fashion :-P

Reply:
ok we did not yet, but at latest when we start implementing. Which is when we agreed on that document, right?

Reply:
If you prefer, I could also change it to already mention `v2.1`. I just figured, we might instead of v2.1 decide to name the next version e.g. `v3`, in which case the documentation would still be correct, without requiring an update :-)

Reply:
I thought the we want to have 2.1 for the difference between v2 name and version are part of the identity and v2.1 only name must be part of the identity.

Or have we agreed on only using the version if the name clashes for current v2 components?